### PR TITLE
network-next (v256)

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by yuwata** [comment]
@mrc0mmand timeout command is broken on Arch Linux??

https://jenkins-systemd.apps.ocp.cloud.ci.centos.org/job/upstream-vagrant-archlinux/9123/artifact/systemd-centos-ci/artifacts_or7fdz1f/vagrant-logs.yn5/vagrant-arch-testsuite.24S/systemd-networkd-tests.py_FAIL.log
```
[TASK START] Wed Jan  3 08:10:08 PM UTC 2024
...
[  190.516420] systemd-networkd[72040]: SELinux enabled state cached to: disabled
...
[ 1469.719174] systemd-networkd[374416]: Bus bus-api-network: changing state RUNNING â†’ CLOSEDok
test_nexthop (__main__.NetworkdNetworkTests.test_nexthop) ... 
  test_nexthop (__main__.NetworkdNetworkTests.test_nexthop) (manage_foreign_nexthops=True) ... FAIL
Command exited with non-zero status 124
	Command being timed: "timeout -k 60s 60m ./test/test-network/systemd-networkd-tests.py"
	User time (seconds): 16.44
	System time (seconds): 13.66
	Percent of CPU this job got: 0%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:00:00
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 40092
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 50
	Minor (reclaiming a frame) page faults: 1683380
	Voluntary context switches: 57196
	Involuntary context switches: 3118
	Swaps: 0
	File system inputs: 121576
	File system outputs: 37584
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 124
[TASK END] Wed Jan  3 10:10:08 PM CET 2024
```

At least, I have no idea about the CI failure in CentOS CI (Arch).

**Comment by mrc0mmand** [comment]
It looks like systemd-networkd SIGABRTs and the test then just hangs, leading up to a timeout:

```
[ 2191.770661] systemd-networkd[365738]: veth99: link_check_ready(): static nexthops are not configured.
[ 2191.770681] systemd-networkd[365738]: veth99: Remembering static nexthop (configured): id: 3, gw: ::, blackhole: no, group: n/a, flags: n/a
[ 2191.770700] systemd-networkd[365738]: veth99: Remembering static nexthop (configured): id: 4, gw: 0.0.0.0, blackhole: no, group: n/a, flags: n/a
[ 2191.770719] systemd-networkd[365738]: veth99: Remembering static nexthop (configured): id: 5, gw: 192.168.10.1, blackhole: no, group: n/a, flags: onlink
[ 2191.770744] systemd-networkd[365738]: veth99: Configuring static route (requesting): dst: 10.10.10.12/32, src: n/a, nexthop: 5, prefsrc: n/a, table: main(254), priority: 0, proto: static, scope: global, type: unicast, flags: n/a
[ 2191.770770] systemd-networkd[365738]: Remembering static nexthop (configured): id: 6, gw: 0.0.0.0, blackhole: yes, group: n/a, flags: n/a
[ 2191.770794] systemd-networkd[365738]: Assertion 'e->p.b.key == i->next_key' failed at src/basic/hashmap.c:614, function hashmap_iterate_in_insertion_order(). Aborting.
```

**Comment by yuwata** [comment]
It seems, for some reason, networkd triggers a kernel bug...

**Comment by LiveFreeAndRoam** [comment]
👏 I'll rerun the tests on Monday and report my findings here.

File: src/network/networkd-ndisc.c [hunk 1]
```diff
@@ -1635,38 +2070,70 @@ static int ndisc_router_handler(Link *link, sd_ndisc_router *rt) {
...
+                /* 4.1. Start DHCPv6 client based on the flag. */
+                r = ndisc_start_dhcp6_client(link, rt);
+                if (r < 0)
+                        return r;
 
-        r = ndisc_router_process_default(link, rt);
-        if (r < 0)
-                return r;
+                /* 4.2. Apply reachable time. */
+                r = ndisc_router_process_reachable_time(link, rt);
```
Comment (LiveFreeAndRoam) [inline src/network/networkd-ndisc.c]: This implementation of Reachable Time is not compliant with IPv6 certification tests.  Please see https://github.com/systemd/systemd/issues/31842.

**Pull request closed without merging**